syntax = "proto3";

package processor;

service Processor {
    rpc ProcessImage (ProcessImageRequest) returns (ProcessImageReply) {}
  }

  message ProcessImageRequest {
    Image image = 1;
  }

  message ProcessImageReply {
    bytes content = 1;
  }

  message Image {
    // Image content, represented as a stream of bytes.
    // Note: As with all `bytes` fields, protobuffers use a pure binary
    // representation, whereas JSON representations use base64.
    bytes content = 1;
    // If both content and source is present use content.
    ImageSource source = 2;
  }

  message ImageSource {
    // The URI of the source document. Can be either:
    //
    // 1. A publicly-accessible image HTTP/HTTPS URL. When fetching images from
    string http_uri = 1;
  }
